vllm/engine/arg_utils.py

```diff
@@ -312,14 +336,24 @@ def create_engine_configs(
         cache_config = CacheConfig(self.block_size,
                                    self.gpu_memory_utilization,
                                    self.swap_space, self.kv_cache_dtype,
-                                   model_config.get_sliding_window(),
-                                   self.enable_prefix_caching)
-        parallel_config = ParallelConfig(self.pipeline_parallel_size,
-                                         self.tensor_parallel_size,
-                                         self.worker_use_ray,
-                                         self.max_parallel_loading_workers,
-                                         self.disable_custom_all_reduce,
-                                         self.ray_workers_use_nsight)
+                                   model_config.get_sliding_window())
+        if self.tokenizer_pool_size:
+            if isinstance(self.tokenizer_pool_extra_config, str):
+                tokenizer_pool_extra_config_parsed = json.loads(
+                    self.tokenizer_pool_extra_config)
+            else:
+                tokenizer_pool_extra_config_parsed = (
+                    self.tokenizer_pool_extra_config or {})
+            tokenizer_pool_config = TokenizerPoolConfig(
+                self.tokenizer_pool_size, self.tokenizer_pool_type,
+                tokenizer_pool_extra_config_parsed)
+        else:
+            tokenizer_pool_config = None
```

Comment:
feels like all this complexity can be put in a helper function, will simplify `create_engine_configs`